core/state/rw_v3.go

```diff
@@ -428,13 +448,49 @@ func (rs *StateV3) writeStateHistory(roTx kv.Tx, txTask *exec22.TxTask, agg *lib
 				}
 			}
 		}
-		if err := agg.AddCodePrev(addr, codePrev); err != nil {
-			return err
-		}
+		//if err := agg.UpdateCode(addr, newCode, codePrev); err != nil {
+		//	return err
+		//}
 	}
 	return nil
 }
 
+func (rs *StateV3) applyUpdates(roTx kv.Tx, task *exec22.TxTask, agg *libstate.AggregatorV3) {
+	//emptyRemoval := task.Rules.IsSpuriousDragon
+	rs.lock.Lock()
+	defer rs.lock.Unlock()
+
+	var p2 []byte
+	for table, wl := range task.WriteLists {
+		var d *libstate.DomainMem
+		switch table {
+		case kv.PlainState:
+			d = rs.shared.Account
+		case kv.Code:
+			d = rs.shared.Code
+		case StorageTable:
+			d = rs.shared.Storage
+		default:
+			panic(fmt.Errorf("unknown table %s", table))
+		}
+
+		for i := 0; i < len(wl.Keys); i++ {
+			addr, err := hex.DecodeString(wl.Keys[i])
+			if err != nil {
+				panic(err)
+			}
+			if len(addr) > 28 {
+				p2 = addr[length.Addr : len(addr)-length.Incarnation]
```

Comment:
1. did you mean: `addr[20+8:]`?
2. probably need use `[]byte(wl.Keys[i])` instead of `hex.DecodeString`